src/configuration/singleton.py

```diff
@@ -0,0 +1,6 @@
+class Singleton(type):
```

Comment:
So I did not want `Configuration.py` to a file of functions because I wanted to read in the config and store it somewhere. That way we reduce amount of IO. The reason I chose to use singleton pattern for two reasons: 1.) We don't need multiple instances of the config object if they will all hold the same data(if that makes sense). For example, Block uses a config instance and so does api. If each of those objects use the exact same data and functionality, then why not make them reference the same instance. Also, this will prevent one instance from having data that another one does not. 2.) We don't have to dependency inject configuration object into all the other classes. Dependency injection is a really nice design pattern but I figured it would be easier to read/understand with the singleton. Also, DI does not protect against other classes creating their own instance of Configuration. With singleton pattern, once an instance is created it will be the only one 
I will move singleton class into configurtion.py